### PR TITLE
[login] check for X-FORWARDED-FOR before REMOTE_ADDR

### PR DESCRIPTION
**Pull request by ridz1208**
### Brief summary of changes
the `user_login_history` table of the database has been logging the IP address of the proxy instead of the real IP of the user. to prevent that, we are checking for an IP address set in the `X-FORWARDED-FOR` variable and only if that is NULL we are fetching the `REMOTE_ADDR` and setting it.

### CODE REVIEW

**Comment by driusan** [comment]
@jkat can you review/test that this fixes your problem with the IP address in the history?

File: php/libraries/SinglePointLogin.class.inc [hunk 1]
```diff
@@ -238,9 +238,8 @@ class SinglePointLogin
                      'Success'         => 'Y',
                      'Failcode'        => null,
                      'Login_timestamp' => date('Y-m-d H:i:s'),
-                     'IP_address'      => isset($_SERVER['REMOTE_ADDR'])
-                                              ? $_SERVER['REMOTE_ADDR']
-                                              : 'Unknown',
+                     'IP_address'      => $_SERVER['HTTP_X_FORWARDED_FOR']
+                         ?? $_SERVER['REMOTE_ADDR'],
```
Comment (driusan) [inline php/libraries/SinglePointLogin.class.inc]:
```suggestion
                         ?? $_SERVER['REMOTE_ADDR'],
                         ?? 'Unknown',
```

Add fallback on unknown back.

Comment (Jkat) [reply php/libraries/SinglePointLogin.class.inc]:
```suggestion
                         ?? $_SERVER['REMOTE_ADDR']
                         ?? 'Unknown',
```

Extra comma in syntax

**Comment by ridz1208** [comment]
@driusan @Jkat done

**Pull request merged**